src/dataset-builder/CohortEditor.ts

```diff
@@ -183,19 +200,25 @@ interface DomainOptionWithKind extends DomainOption {
   kind: 'domain';
 }
 
-type CriteriaOption = DomainOptionWithKind | ProgramDataRangeOption | ProgramDataListOption;
+type CriteriaOption = DomainOptionWithKind | SnapshotBuilderProgramDataOption;
 
-export function criteriaFromOption(option: DomainOptionWithKind): undefined;
-export function criteriaFromOption(option: ProgramDataListOption): ProgramDataListCriteria;
-export function criteriaFromOption(option: ProgramDataRangeOption): ProgramDataRangeCriteria;
-export function criteriaFromOption(option: CriteriaOption): AnyCriteria;
+export function criteriaFromOption(datasetId: string, index: number, option: DomainOptionWithKind): Promise<undefined>;
+export function criteriaFromOption(
+  datasetId: string,
+  index: number,
+  option: SnapshotBuilderProgramDataOption
+): Promise<ProgramDataListCriteria | ProgramDataRangeCriteria>;
+export function criteriaFromOption(datasetId: string, index: number, option: CriteriaOption): Promise<AnyCriteria>;
 
-export function criteriaFromOption(option: CriteriaOption): AnyCriteria | undefined {
+export async function criteriaFromOption(
+  datasetId: string,
+  index: number,
+  option: CriteriaOption
+): Promise<AnyCriteria | undefined> {
   switch (option.kind) {
     case 'list':
-      return createDefaultListCriteria(option);
     case 'range':
-      return createDefaultRangeCriteria(option);
+      return createDefaultProgramDataCriteria(datasetId, index, option);
     case 'domain':
```

Comment:
It's not handled here because this is only used to create a criteria from program data options. The code to create domain critera from an option (aka concept) is the function `toCriteria()` in `DomainCriteriaSelector.ts`. I can look at trying to make the types reflect this.